src/3d/qgs3dmapscene.cpp

```diff
@@ -760,6 +808,17 @@ void Qgs3DMapScene::removeLayerEntity( QgsMapLayer *layer )
     mChunkEntities.removeOne( chunkedEntity );
   }
 
+  if ( QgsVirtualPointCloudEntity *vpcNewEntity = qobject_cast<QgsVirtualPointCloudEntity *>( entity ) )
```

Comment:
Again, I think this should live in a virtual method in the entity rather then in this generic class